acmOpportunities/main.py

```diff
@@ -0,0 +1,434 @@
+        "X-RapidAPI-Host": rapid_api_host,
+    }
+
+    rapid_jobs = []
+    response = requests.get(url, headers=headers).json()
+
+    command_line_value = extract_command_value()  # Extracts command-line value
+
+    for elem in response["hits"]:
+        time = elem["formatted_relative_time"]
+
+        numeric = re.search(r"\d+", time)
+        formatted_time_integer = int(numeric.group()) if numeric else 0
+
+        if len(rapid_jobs) <= 5 and int(command_line_value) >= formatted_time_integer:
+            job = {}
+            job["_company"] = elem["company_name"]
+            job["_title"] = elem["title"]
+            job["_location"] = elem["location"]
+            job["_link"] = f'https://www.indeed.com/viewjob?jk={elem["id"]}&locality=us'
+
+        # print(elem)
+
+        rapid_jobs.append(job)
+
+    return rapid_jobs
+
+
+# -------------------FOR LINKEDIN----------------------------
+
+
+def linkedin_response() -> List[object]:
+    url = os.getenv("LINKEDIN_URL")
+
+    response = requests.get(url)
+
+    content = response.text
+
+    parse_content = BeautifulSoup(content, "html.parser")
+
+    div_post = parse_content.find_all(
+        "div",
+        class_="base-card relative w-full hover:no-underline focus:no-underline base-card--link base-search-card base-search-card--link job-search-card",
+    )
+
+    date_div_post = parse_content.find_all(
+        "time",
+        class_="job-search-card__listdate",
+    )  # Recieves all the job posting dates
+
+    command_line_value = extract_command_value()  # Extracts command-line value
+
+    linked_in_jobs = []
+
+    for elem, date in zip(div_post, date_div_post):
+        # If the value of date.text.strip() has the word "days" in it, we access the number, and compare it with the
+        # value of the command line
+
+        if "days" in date.text.strip():
+            numeric = re.search(r"\d+", date.text.strip())
+            formatted_time_integer = int(numeric.group()) if numeric else 0
+
+            if (
+                len(linked_in_jobs) <= 5
+                and int(command_line_value) >= formatted_time_integer
+            ):
+                job = {}
+                job["_company"] = elem.find(
+                    "h4", class_="base-search-card__subtitle"
+                ).text.strip()
+                job["_title"] = elem.find(
+                    "h3", class_="base-search-card__title"
+                ).text.strip()
+                job["_location"] = elem.find(
+                    "span", class_="job-search-card__location"
+                ).text.strip()
+                job["_link"] = elem.find("a", class_="base-card__full-link")[
+                    "href"
+                ].split("?")[0]
+
+                linked_in_jobs.append(job)
+
+    return linked_in_jobs
+
+
+# -------------------FOR HELPER FUNCTIONS ----------------------------
+
+
+def shorten_linkedin_list():
+    connection = sqlite3.connect(job_db)
+    cursor = connection.cursor()
```

Comment:
I suggest switching to `with` statements so that using connection can be cleaned up and look similar to:

```python
with sqlite3.connect(job_db) as conn:
    # stuff
    conn.commit()
```